src/app/clusters/zone-management-server/zone-management-server.h

```diff
@@ -0,0 +1,356 @@
+
+protected:
+    ZoneMgmtServer * GetZoneMgmtServer() const { return mZoneMgmtServer; }
+};
+
+enum class OptionalAttribute : uint32_t
+{
+};
+
+class ZoneMgmtServer : public CommandHandlerInterface, public AttributeAccessInterface
+{
+public:
+    /**
+     * @brief Creates a Zone Management cluster instance. The Init() function needs to be called for this instance
+     * to be registered and called by the interaction model at the appropriate times.
+     *
+     * @param aDelegate                         A pointer to the delegate to be used by this server.
+     *                                          Note: the caller must ensure that the delegate lives throughout the instance's
+     *                                          lifetime.
+     *
+     * @param aEndpointId                       The endpoint on which this cluster exists. This must match the zap configuration.
+     * @param aFeatures                         The bitflags value that identifies which features are supported by this instance.
+     * @param aOptionalAttrs                    The bitflags value that identifies the optional attributes supported by this
+     *                                          instance.
+     * @param aMaxUserDefinedZones              The maximum number of user-defined zones supported by the device.
+     *                                          This value is specified by the device manufacturer.
+     * @param aMaxZones                         The maximum number of zones that are allowed to exist on the device. This is the
+     *                                          sum of the predefined built-in zones and the user-defined zones.
+     * @param aSensitivityMax                   The hardware-specific value for the number of supported sensitivity levels.
+     *                                          This value is specified by the device manufacturer.
+     * @param aTwoDCartesianMax                 The maximum X and Y points that are allowed for TwoD Cartesian Zones.
+     *
+     */
+    ZoneMgmtServer(ZoneMgmtDelegate & aDelegate, EndpointId aEndpointId, const BitFlags<Feature> aFeatures,
+                   const BitFlags<OptionalAttribute> aOptionalAttrs, uint8_t aMaxUserDefinedZones, uint8_t aMaxZones,
+                   uint8_t aSensitivityMax, const TwoDCartesianVertexStruct & aTwoDCartesianMax);
+
+    ~ZoneMgmtServer() override;
+
+    /**
+     * @brief Initialise the Zone Management server instance.
+     * This function must be called after defining a ZoneMgmtServer class object.
+     * @return Returns an error if the given endpoint and cluster ID have not been enabled in zap or if the
+     * CommandHandler or AttributeHandler registration fails, else returns CHIP_NO_ERROR.
+     * This method also checks if the feature setting is valid, if invalid it will return CHIP_ERROR_INVALID_ARGUMENT.
+     */
+    CHIP_ERROR Init();
```

Comment:
This seems to assume that any mfg zones are added before calling Init()?  Trying to understand how that Init check works if that's not the case.

Reply:
On second thoughts, I felt it is best to remove this mfg-zone check in Init to allow for the flexibility of defining mfg zones later.